### DISCUSSION TITLE
Weird problems with hanging queries

### DISCUSSION THREAD

**Discussion by czocher** [maxcountryman/tower-sessions#99] [Q&A]
When using the `PostgresStore` the session reads seem to hang indefinitely after a few moments.  It does not happen with the `MemoryStore`.

This is what `select * from pg_stat_activity` returns:

```csv
datid,datname,pid,leader_pid,usesysid,usename,application_name,client_addr,client_hostname,client_port,backend_start,xact_start,query_start,state_change,wait_event_type,wait_event,state,backend_xid,backend_xmin,query_id,query,backend_type
16384,ferret,54,NULL,10,ferret,,10.89.5.7,NULL,50488,2023-11-17 15:56:01.469687+00,NULL,2023-11-17 15:56:16.208738+00,2023-11-17 15:56:16.209205+00,Client,ClientRead,idle,NULL,NULL,NULL,"select data from ""tower_sessions"".""session"" where id = $1 and expiry_date > $2 ",client backend
```

I believe `tokio` cancels the task for some reason, leaving it hanging on the database side. I'm pretty baffled by why it happens. Any ideas what could I try to debug this issue?

**Comment by maxcountryman** [comment]
Could it be related to how the pool is configured? Perhaps that's something you could experiment with.

I'd also be curious to see if Postgres is holding onto a lock internally: maybe Tokio is cancelling the task because it's become too slow waiting for Postgres to release a row lock? (Maybe something like this query could be helpful: https://stackoverflow.com/a/72374897/1542038)

**Comment by maxcountryman** [comment]
I've removed the shared memory caching in the latest release: this seems to have caused issues with other backends, so while it might be entirely unrelated, I'm also curious if you see this issue on the 0.6.0 series?

**Comment by czocher** [comment]
Thanks @maxcountryman, I updated the crate, it seems to work perfectly now!
